src/aliased_buffer.h

```diff
@@ -141,6 +141,30 @@ class AliasedBuffer {
       return aliased_buffer_->GetValue(index_);
     }
 
+    template <typename T>
+    inline Reference& operator+=(const T& val) {
+      const T current = aliased_buffer_->GetValue(index_);
+      aliased_buffer_->SetValue(index_, current + val);
+      return *this;
+    }
+
+    inline Reference& operator+=(const Reference& val) {
+      this->operator+=(static_cast<NativeT>(val));
+      return *this;
+    }
+
+    template <typename T>
+    inline Reference& operator-=(const T& val) {
+      const T current = aliased_buffer_->GetValue(index_);
+      aliased_buffer_->SetValue(index_, current - val);
+      return *this;
+    }
+
+    inline Reference& operator-=(const Reference& val) {
+      this->operator-=(static_cast<NativeT>(val));
```

Comment:
Just a suggestion, but I would personally find `return *this -= static_cast<NativeT>(val);` a bit clearer/more idiomatic in these places.

Reply:
Ah yes, let me change that thanks!